unbxdSearch.js

```diff
@@ -1078,7 +1175,11 @@ var unbxdSearchInit = function(jQuery, Handlebars) {
             } else if (this.options.type == "browse" && this.params['categoryId'] != undefined) {
                 url += '&category-id=' + encodeURIComponent(this.params.categoryId);
             } else if (isTypeCategory(this.options.type) && this.params.categoryId !== undefined) {
-                url += "&p=" + encodeURIComponent(this.params.categoryId);
+                url += "&p=" + this.options.mappedFields.categoryField + ":\"" + encodeURIComponent(this.params.categoryId) + "\"" + '&pagetype=boolean';
+            }
+
+            if (this.params.hasOwnProperty('categoryFilter') && this.params.categoryFilter !== '') {
```

Comment:
```suggestion
            if (this.params.hasOwnProperty('categoryFilter') && this.params.categoryFilter) {
```

Will handle `null` or `undefined`.

Reply:
Yes, this.params.categoryFilter will return false if any of "",undefined or null is the value

Reply:
I'm suggesting a change here :). With the current setup it wont handle `undefined or null`. You need to remove `!== ''`

Reply:
Done